### PR TITLE
Rename Azul storage buckets (#4646)

### LINKED ISSUE #4646: Rename Azul storage buckets
Current storage bucket names are wildly inconsistent:

- `edu-ucsc-gi-singlecell-azul-storage-dev`
- `edu-ucsc-gi-singlecell-azul-storage-sandbox`
- `edu-ucsc-gi-azul-dcp2-prod-storage-prod`
- `edu-ucsc-gi-platform-anvil-dev-anvilbox`
- `edu-ucsc-gi-platform-anvil-dev-anvildev`
- `edu-ucsc-gi-platform-anvil-prod-anvilprod`

We want them to be consistent so that we can switch to using the `qualified_bucket_name()` config method to generated them, in #4655. 

Change the values of the AZUL_S3_BUCKET variable for all deployments so that it match the template `edu-ucsc-gi-{aws.accountname}-storage-{deployment_stage}.{aws.region}`. The storage buckets contain cached manifests, lambda layers and cached health statuses. None of that information must be retained and is recreated automatically when needed, albeit at a slight performance cost. 

During deployment, a new bucket should be created under the new name. The old bucket should be left in place by Terraform and deleted later as a manual upgrade step. By removing the bucket resource from Terraform state, Terraform can be tricked into believing that the bucket does not exist, so that it creates a new one, using the new name. Resources can be removed from TF state with `rename_resources.py`.

### PR DESCRIPTION
**Pull request by nadove-ucsc**
Connected issues: #4646

### Author

- [x] PR is a draft
- [x] Target branch is `develop`
- [x] Name of PR branch matches `issues/<GitHub handle of author>/<issue#>-<slug>`
- [x] PR title references all connected issues
- [x] PR title matches<sup>1</sup> that of a connected issue <sub>or comment in PR explains why they're different</sub>
- [x] For each connected issue, there is at least one commit whose title references that issue
- [x] PR is connected to all connected issues via ZenHub
- [x] PR description links to connected issues
- [x] Added `partial` label to PR <sub>or this PR completely resolves all connected issues</sub>

<sup>1</sup> when the issue title describes a problem, the corresponding PR
title is `Fix: ` followed by the issue title

### Author (reindex, API changes)

- [x] Added `r` tag to commit title <sub>or this PR does not require reindexing</sub>
- [x] Added `reindex` label to PR <sub>or this PR does not require reindexing</sub>
- [x] Added `a` (compatible changes) or `A` (incompatible ones) tag to commit title <sub>or this PR does not modify the Azul service API</sub>
- [x] Added `API` label to connected issues <sub>or this PR does not modify the Azul service API</sub>

### Author (chains)

- [x] This PR is blocked by previous PR in the chain <sub>or this PR is not chained to another PR</sub>
- [x] Added `base` label to the blocking PR <sub>or this PR is not chained to another PR</sub>
- [x] Added `chained` label to this PR <sub>or this PR is not chained to another PR</sub>

### Author (upgrading)

- [x] Documented upgrading of deployments in UPGRADING.rst <sub>or this PR does not require upgrading</sub>
- [x] Added `u` tag to commit title <sub>or this PR does not require upgrading</sub>
- [x] Added `upgrade` label to PR <sub>or this PR does not require upgrading</sub>

### Author (operator tasks)

- [x] Added checklist items for additional operator tasks <sub>or this PR does not require additional tasks</sub>

### Author (hotfixes)

- [x] Added `F` tag to main commit title <sub>or this PR does not include permanent fix for a temporary hotfix</sub>
- [x] Reverted the temporary hotfixes for any connected issues <sub>or the `prod` branch has no temporary hotfixes for any connected issues</sub>

### Author (before every review)

- [x] Rebased PR branch on `develop`, squashed old fixups
- [x] Ran `make requirements_update` <sub>or this PR does not touch requirements*.txt, common.mk, Makefile and Dockerfile</sub>
- [x] Added `R` tag to commit title <sub>or this PR does not touch requirements*.txt</sub>
- [x] Added `reqs` label to PR <sub>or this PR does not touch requirements*.txt</sub>
- [x] `make integration_test` passes in personal deployment <sub>or this PR does not touch functionality that could break the IT</sub>

### Peer reviewer (after requesting changes)

Uncheck the *Author (before every review)* checklists.

### Peer reviewer (after approval)

- [x] PR is not a draft
- [x] Ticket is in *Review requested* column
- [x] Requested review from primary reviewer
- [x] Assigned PR to primary reviewer

### Primary reviewer (after requesting changes)

Uncheck the *before every review* checklists. Update the `N reviews` label.

### Primary reviewer (after approval)

- [x] Actually approved the PR
- [x] Labeled connected issues as `demo` or `no demo`
- [x] Commented on connected issues about demo expectations <sub>or all connected issues are labeled `no demo`</sub>
- [x] Decided if PR can be labeled `no sandbox`
- [x] PR title is appropriate as title of merge commit
- [x] `N reviews` label is accurate
- [x] Moved ticket to *Approved* column
- [x] Assigned PR to current operator

### Operator (before pushing merge the commit)

- [x] Checked `reindex` label and `r` commit title tag
- [x] Checked that demo expectations are clear <sub>or all connected issues are labeled `no demo`</sub>
- [x] Squashed PR branch and rebased onto `develop`
- [x] Sanity-checked history
- [x] Pushed PR branch to GitHub
- [x] Pushed PR branch to GitLab `dev` and added `sandbox` label <sub>or PR is labeled `no sandbox`</sub>
- [x] Pushed PR branch to GitLab `anvildev` <sub>or PR is labeled `no sandbox`</sub>
- [x] Build passes in `sandbox` deployment <sub>or PR is labeled `no sandbox`</sub>
- [x] Build passes in `anvilbox` deployment <sub>or PR is labeled `no sandbox`</sub>
- [x] Reviewed build logs for anomalies in `sandbox` deployment <sub>or PR is labeled `no sandbox`</sub>
- [x] Reviewed build logs for anomalies in `anvilbox` deployment <sub>or PR is labeled `no sandbox`</sub>
- [x] Deleted unreferenced indices in `sandbox` <sub>or this PR does not remove catalogs or otherwise causes unreferenced indices </sub>
- [x] Deleted unreferenced indices in `anvilbox` <sub>or this PR does not remove catalogs or otherwise causes unreferenced indices </sub>
- [x] Started reindex in `sandbox` <sub>or this PR does not require reindexing `sandbox`</sub>
- [x] Started reindex in `anvilbox` <sub>or this PR does not require reindexing `sandbox`</sub>
- [x] Checked for failures in `sandbox` <sub>or this PR does not require reindexing `sandbox`</sub>
- [x] Checked for failures in `anvilbox` <sub>or this PR does not require reindexing `sandbox`</sub>
- [x] Added PR reference to merge commit title
- [x] Collected commit title tags in merge commit title
- [x] Moved connected issues to Merged column in ZenHub
- [x] Pushed merge commit to GitHub

### Operator (after pushing the merge commit)

- [x] Shortened the PR chain <sub>or this PR is not labeled `base`</sub>
- [x] Pushed merge commit to GitLab `dev` <sub>or PR is labeled `no sandbox`</sub>
- [x] Pushed merge commit to GitLab `anvildev` <sub>or PR is labeled `no sandbox`</sub>
- [x] Build passes on GitLab `dev`<sup>1</sup>
- [x] Manually deleted old storage buckets for `sandbox` and `dev` 
- [x] Made announcement on slack for other developers to upgrade personal deployments
- [x] Reviewed build logs for anomalies on GitLab `dev`<sup>1</sup>
- [x] Build passes on GitLab `anvildev`<sup>1</sup>
- [x] Manually deleted old storage buckets in `anvilbox` and `anvildev`
- [x] Reviewed build logs for anomalies on GitLab `anvildev`<sup>1</sup>
- [x] Deleted PR branch from GitHub
- [x] Deleted PR branch from GitLab `dev`
- [x] Deleted PR branch from GitLab `anvildev`

<sup>1</sup> When pushing the merge commit is skipped due to the PR being
labelled `no sandbox`, the next build triggered by a PR whose merge commit *is*
pushed determines this checklist item.

### Operator (reindex)

- [x] Deleted unreferenced indices in `dev` <sub>or this PR does not remove catalogs or otherwise causes unreferenced indices </sub>
- [x] Deleted unreferenced indices in `anvildev` <sub>or this PR does not remove catalogs or otherwise causes unreferenced indices </sub>
- [x] Started reindex in `dev` <sub>or this PR does not require reindexing</sub>
- [x] Started reindex in `anvildev` <sub>or this PR does not require reindexing</sub>
- [x] Checked for and triaged indexing failures in `dev` <sub>or this PR does not require reindexing</sub>
- [x] Checked for and triaged indexing failures in `anvildev` <sub>or this PR does not require reindexing</sub>
- [x] Emptied fail queues in `dev` deployment <sub>or this PR does not require reindexing</sub>
- [x] Emptied fail queues in `anvildev` deployment <sub>or this PR does not require reindexing</sub>

### Operator

- [x] Added checklist item to prod promotion PR: "Manually deleted old storage bucket for prod"
- [x] Unassigned PR

## Shorthand for review comments

- `L` line is too long
- `W` line wrapping is wrong
- `Q` bad quotes
- `F` other formatting problem

### CODE REVIEW

**Review by dsotirho-ucsc — DISMISSED** [review]
I think it'd be helpful to the operator if the UPGRADING instructions listed the old bucket names/patterns that should be deleted, however they can view the PR's diff for this info I suppose.

**Review by hannes-ucsc — DISMISSED** [review]

File: UPGRADING.rst [hunk 1]
```diff
@@ -10,6 +10,42 @@ branch that does not have the listed changes, the steps would need to be
...
+``"platform-hca-dev"``. As always, use the sandbox deployment's
+``environment.py`` as a model when upgrading personal deployments.
+
+After the changes are deployed to a given personal deployment, manually delete
+the old storage bucket for that deployment:
+
+1) Open the AWS console and navigate to the S3 mamangement console.
+
+2) Select the old storage bucket for your deployment. Click the "Empty" action.
+Empty the bucket and click "Exit".
```
Comment (hannes-ucsc) [inline UPGRADING.rst]: This renders wrong: image

I use PyCharm's rst preview pane to check for these types of things.

Comment (hannes-ucsc) [reply UPGRADING.rst]: Then again, people should know how to delete buckets. Maybe we don't need this.

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]
One more thing: Since there is nothing that would fail if a personal deployment wasn't upgraded, we should include a checklist for announcement by the operator so that people upgrade their personals immediately.

**Review by hannes-ucsc — APPROVED** [review]

**Pull request merged**